az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -433,52 +442,65 @@ public Void run() throws Exception {
    *
    * @return a Set. May be empty, but will never be null
    */
-  public static Set<String> findApplicationIdFromLog(String logFilePath, Logger log) {
-
-    File logFile = new File(logFilePath);
+  public static Set<String> findApplicationIdFromLog(final String logFilePath, final Logger log) {
+    // At least one job log file must be there.
+    final File logFile = new File(logFilePath);
 
     if (!logFile.exists()) {
       throw new IllegalArgumentException("the logFilePath does not exist: " + logFilePath);
     }
-    if (!logFile.isFile()) {
-      throw new IllegalArgumentException("the logFilePath specified  is not a valid file: "
-          + logFilePath);
-    }
-    if (!logFile.canRead()) {
-      throw new IllegalArgumentException(
-          "unable to read the logFilePath specified: " + logFilePath);
-    }
+
+    // Get the directory and file name
+    final int lastSlash = logFilePath.lastIndexOf('/');
+    final String dirPath = logFilePath.substring(0, lastSlash);
+    final String logFileName = logFilePath.substring(lastSlash + 1);
+
+    final File[] logFiles =
+        new File(dirPath).listFiles((dir, name) -> name.startsWith(logFileName));
 
```

Comment:
Can you add a comment with few files names as an example so it becomes clear what this code does?